pyiron_base/project/jobloader.py

```diff
@@ -0,0 +1,120 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+"""
+A helper class to be assigned to the project, which facilitates tab-completion when
+loading jobs.
+"""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from pyiron_base.state import state
+from pyiron_base.database.jobtable import get_job_id
+from pyiron_base.jobs.job.util import _get_safe_job_name
+
+if TYPE_CHECKING:
+    from pyiron_base.jobs.job.generic import GenericJob
+    from pyiron_base.jobs.job.path import JobPath
+    from pyiron_base.project.generic import Project
+
+
+class _JobByAttribute(ABC):
+    """
+    A parent class for accessing project jobs by a call and a job specifier, or by tab
+    completion.
+    """
+
+    def __init__(self, project: Project):
+        self._project = project
+
+    @property
+    def _job_table(self):
+        return self._project.job_table(recursive=False, columns=["job"])
```

Comment:
Why is this `recursive=False` ?

Reply:
It was intended to make sure that the mapping between the tab-completion menu and actual jobs stayed injective. I see now that the load function already handles duplicate job names at different levels by returning the upper-most job. I'm removing the flag specification here so that the tab-completion behaves the same as the string-based loading.